src/AbstractPhpdocToTypeDeclarationFixer.php

```diff
@@ -29,6 +29,8 @@
 
 /**
  * @internal
+ *
+ * @phpstan-type _CommonTypeInfo array{commonType: string, isNullable: bool}
```

Comment:
> you said you are against

Time passed, i believe this is not the only element i adjusted my mind on, and you managed to convince me of few other topics by now too.
I would still prefer to have lightweight php-native type aliasing, or proper types, but it could be overkill preventing to fix whole codebase fast. Ultimately I would still love to have native types than phpstan aliases.

> _

Anything to determine it's "special", to easily identify it for myself when introducing proper DTOs or to prevent conversion from phpdoc to method signature.

Reply:
> Anything to determine it's "special", to easily identify it for myself when introducing proper DTOs

Isn't `@phpstan-type` enough?

Reply:
for console grep both approaches are valid.

for experimental `@annotation` to type, `_` handling is somehow easier to implement for myself (we in-place add starts_with(_) check rather than checking is this phpdoc declares this type? or maybe type was declared on class docs? let's look in few places)

For me it's a mid-path to go till achieving proper solution (native type that will crash execution/compliation, if violated). For that, I do not put attention to this detail that much and with pragmatic approach, the discussion is taking more effort than value of going in any of those direction, for myself.

Reply:
`_` removed

Reply:
reverted :(

PHPStan likes `DocumentElement&array{type: 'property'}` , but not recognising union type on `array{type: 'property'}&DocumentElement` (I believe bug on stan?)

same time, our own rules are asking `B&A` to be re-sorted into `A&B`